tests/lib/openstack_query/handlers/test_server_side_handler.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from parameterized import parameterized
+
+from openstack_query.handlers.server_side_handler import ServerSideHandler
+from exceptions.query_preset_mapping_error import QueryPresetMappingError
+
+from tests.lib.openstack_query.mocks.mocked_query_presets import MockQueryPresets
+from tests.lib.openstack_query.mocks.mocked_props import MockProperties
+
+
+class ServerSideHandlerTests(unittest.TestCase):
+    """
+    Runs various tests to ensure that ServerSideHandler class methods function expectedly
+    """
+
+    def setUp(self):
+        """
+        Setup for tests
+        """
+        super().setUp()
+
+        _SERVER_SIDE_FUNCTION_MAPPINGS = {
+            MockQueryPresets.ITEM_1: {
+                MockProperties.PROP_1: "item1-prop1-kwarg",
+                MockProperties.PROP_2: "item2-prop2-kwarg",
+            },
+            MockQueryPresets.ITEM_2: {
+                MockProperties.PROP_3: "item2-prop3-kwarg",
+                MockProperties.PROP_4: "item2-prop4-kwarg",
+            },
+        }
+        self.instance = ServerSideHandler(_SERVER_SIDE_FUNCTION_MAPPINGS)
+
+    def test_check_supported_true(self):
+        self.assertTrue(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_1, MockProperties.PROP_1
+            )
+        )
+
+    def test_check_supported_false(self):
+        # when preset is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_3, MockProperties.PROP_1
+            )
+        )
+
+        # when preset found, but prop is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_1, MockProperties.PROP_3
+            )
+        )
+
+    def test_get_mapping_valid(self):
+        self.assertEqual(
+            self.instance._get_mapping(MockQueryPresets.ITEM_1, MockProperties.PROP_1),
+            "item1-prop1-kwarg",
+        )
+
+    def test_get_mapping_invalid(self):
+        # when preset is not supported
+        self.assertIsNone(
+            self.instance._get_mapping(MockQueryPresets.ITEM_3, MockProperties.PROP_1)
+        )
+
+        # when preset found, but prop is not supported
+        self.assertIsNone(
+            self.instance._get_mapping(MockQueryPresets.ITEM_1, MockProperties.PROP_3)
+        )
+
+    @patch(
+        "openstack_query.handlers.server_side_handler.ServerSideHandler._check_filter_mapping"
+    )
+    @patch(
+        "openstack_query.handlers.server_side_handler.ServerSideHandler._get_mapping"
+    )
+    def test_get_filters_valid(self, mock_get_mapping, mock_check_filter_mapping):
+        mock_params = {"arg1": "val1", "arg2": "val2"}
+        mock_filter_func = MagicMock()
```

Comment:
Do we have any tests for our filter funcs separately?

Reply:
yes - but filter funcs are handled in `ClientSideHandlers` - what are you referring to here? 
This function checks if a set of kwargs are returned that we can pass to openstacksdk command